homeassistant/components/devolo_home_network/device_tracker.py

```diff
@@ -0,0 +1,162 @@
+"""Platform for device tracker integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from devolo_plc_api.device import Device
+
+from homeassistant.components.device_tracker import (
+    DOMAIN as DEVICE_TRACKER_DOMAIN,
+    SOURCE_TYPE_ROUTER,
+)
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DATA_RATE_MEGABITS_PER_SECOND,
+    FREQUENCY_GIGAHERTZ,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    CONNECTED_STATIONS,
+    CONNECTED_WIFI_CLIENTS,
+    DOMAIN,
+    MAC_ADDRESS,
+    WIFI_APTYPE,
+    WIFI_BANDS,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all devices and sensors and setup them via config entry."""
+    device: Device = hass.data[DOMAIN][entry.entry_id]["device"]
+    coordinators: dict[str, DataUpdateCoordinator] = hass.data[DOMAIN][entry.entry_id][
+        "coordinators"
+    ]
+    registry = entity_registry.async_get(hass)
+    tracked = set()
+
+    @callback
+    def new_device_callback() -> None:
+        """Add new devices if needed."""
+        new_entities = []
+        for station in coordinators[CONNECTED_WIFI_CLIENTS].data[CONNECTED_STATIONS]:
+            if station[MAC_ADDRESS] in tracked:
+                continue
+
+            new_entities.append(
+                DevoloScannerEntity(
+                    coordinators[CONNECTED_WIFI_CLIENTS], device, station[MAC_ADDRESS]
+                )
+            )
+            tracked.add(station[MAC_ADDRESS])
+            if new_entities:
+                async_add_entities(new_entities)
+
+    @callback
+    def restore_entities() -> None:
+        """Restore clients that are not a part of active clients list."""
+        missing = []
+        for entity in registry.entities.values():
+            mac_address = entity.unique_id.replace(f"{device.serial_number}_", "")
+            if (
+                entity.config_entry_id == entry.entry_id
+                and entity.platform == DOMAIN
+                and entity.domain == DEVICE_TRACKER_DOMAIN
+                and mac_address not in tracked
+            ):
+                missing.append(
+                    DevoloScannerEntity(
+                        coordinators[CONNECTED_WIFI_CLIENTS], device, mac_address
+                    )
+                )
+                tracked.add(mac_address)
+
+        if missing:
+            async_add_entities(missing)
+
+    if device.device and "wifi1" in device.device.features:
```

Comment:
You shouldn't forward the setup of this platform unless this check is true.

Reply:
As the devolo devices have different capabilities, this could be done for all the other platforms as well and would need some extra logic to calculate which platform is suitable. Do you mind, if we handle that in a separate MR?